### PR TITLE
Session recording

### PR DESCRIPTION
**Pull request by iainx**
Implement recording of key events and commands executed. Save them to an XML file and replay them at a later date

### CODE REVIEW

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestSession.cs [hunk 1]
```diff
@@ -337,7 +343,15 @@ public AppResult[] ExecuteQuery (AppQuery query, int timeout = 20000)
 			} catch (TimeoutException e) {
 				throw new TimeoutException (string.Format ("Timeout while executing ExecuteQuery: {0}", query), e);
 			}
+/*
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestSession.cs]: Are these comments necessary?

Comment (iainx) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Components.AutoTest/AutoTestSession.cs]: Probably not :)

**Comment by Therzok** [comment]
:+1:

**Comment by iainx** [comment]
Removed commented code and rebased

**Pull request merged**